File: lib/screens/nova_screen.dart
import 'dart:convert';
import 'dart:ui';
import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';
import 'package:shared_preferences/shared_preferences.dart';
import 'package:http/http.dart' as http;
import 'package:flutter_animate/flutter_animate.dart';
import '../theme/colors.dart';
import '../widgets/glass_card.dart';
import '../widgets/nova_orb.dart';

class NovaScreen extends StatefulWidget {
  const NovaScreen({Key? key}) : super(key: key);

  @override
  State<NovaScreen> createState() => _NovaScreenState();
}

class _NovaScreenState extends State<NovaScreen> {
  final TextEditingController _inputController = TextEditingController();
  final ScrollController _scrollController = ScrollController();

  List<Map<String, String>> _messages = [
    {
      'id': '1',
      'from': 'nova',
      'text': "I noticed you've been feeling overwhelmed this week. I'm here for you. What would you like to do?",
    }
  ];

  bool _loading = false;
  String _apiKey = '';
  bool _safetyTriggered = false;
  NovaState _novaState = NovaState.idle;

  final List<String> _crisisWords = [
    "suicide", "kill myself", "harm myself", "end my life", "want to die", "self harm",
  ];

  @override
  void initState() {
    super.initState();
    _loadApiKey();
  }

  Future<void> _loadApiKey() async {
    final prefs = await SharedPreferences.getInstance();
    setState(() {
      _apiKey = prefs.getString('zenara_openrouter_api_key') ?? '';
    });
  }

  bool _checkForCrisis(String text) {
    final lower = text.toLowerCase();
    return _crisisWords.any((word) => lower.contains(word));
  }

  void _updateNovaState(String text, bool isTyping) {
    if (_safetyTriggered) {
      _novaState = NovaState.stressed;
      return;
    }
    
    if (isTyping) {
      _novaState = NovaState.typing;
      return;
    }

    if (_loading) {
      _novaState = NovaState.thinking;
      return;
    }

    final lower = text.toLowerCase();
    if (lower.contains('happy') || lower.contains('good') || lower.contains('great')) {
      _novaState = NovaState.happy;
    } else if (lower.contains('stress') || lower.contains('overwhelmed') || lower.contains('anxious')) {
      _novaState = NovaState.stressed;
    } else if (lower.contains('sleep') || lower.contains('tired')) {
      _novaState = NovaState.sleeping;
    } else {
      _novaState = NovaState.idle;
    }
  }

  String _getMockResponse(String userInput) {
    final lower = userInput.toLowerCase();
    if (lower.contains("reflect")) {
      return "Let's take a moment to look back at your day. What was one moment where you felt completely in control, however small?";
    }
    if (lower.contains("reframe")) {
      return "I can help with that. Share a negative thought you've had today, and we'll work together to identify cognitive distortions and reframe it gently.";
    }
    if (lower.contains("breathe") || lower.contains("breathing")) {
      return "Let's do a quick Box Breathing exercise. Inhale for 4 seconds, hold for 4, exhale for 4, hold for 4. Shall we begin?";
    }
    if (lower.contains("hello") || lower.contains("hi") || lower.contains("hey")) {
      return "Hello! I'm Nova, your emotional companion. How is your heart feeling today?";
    }
    return "I hear you, and I'm holding space for you. Thank you for sharing that with me. What feelings are surfacing for you right now?";
  }

  Future<void> _sendMessage([String? textToSend]) async {
    final text = textToSend ?? _inputController.text;
    if (text.trim().isEmpty || _loading || _safetyTriggered) return;

    if (_checkForCrisis(text)) {
      setState(() {
        _safetyTriggered = true;
        _novaState = NovaState.stressed;
        _messages.add({
          'id': DateTime.now().millisecondsSinceEpoch.toString(),
          'from': 'user',
          'text': text,
        });
        _messages.add({
          'id': (DateTime.now().millisecondsSinceEpoch + 1).toString(),
          'from': 'nova',
          'text': "Safety Warning: I have detected indicators of crisis. The AI companion is suspended. Please call 988 immediately.",
        });
        _inputController.clear();
      });
      _scrollToEnd();
      return;
    }

    setState(() {
      _messages.add({
        'id': DateTime.now().millisecondsSinceEpoch.toString(),
        'from': 'user',
        'text': text,
      });
      if (textToSend == null) {
        _inputController.clear();
      }
      _loading = true;
      _updateNovaState(text, false);
    });
    _scrollToEnd();

    if (_apiKey.isNotEmpty) {
      try {
        final systemPrompt = "You are Nova, Zenara's compassionate AI mental health companion.";
        final response = await http.post(
          Uri.parse('https://openrouter.ai/api/v1/chat/completions'),
          headers: {
            'Content-Type': 'application/json',
            'Authorization': 'Bearer $_apiKey',
          },
          body: json.encode({
            'model': 'anthropic/claude-3.5-sonnet',
            'messages': [
              {'role': 'system', 'content': systemPrompt},
              ..._messages.where((m) => m['from'] != 'nova' || !m['text']!.startsWith("Safety Warning")).map((m) => {
                'role': m['from'] == 'user' ? 'user' : 'assistant',
                'content': m['text'],
              }).toList(),
            ],
          }),
        );

        if (response.statusCode == 200) {
          final data = json.decode(response.body);
          String reply = "I'm here with you. Tell me more.";
          if (data['choices'] != null && data['choices'].isNotEmpty) {
            reply = data['choices'][0]['message']['content'] ?? reply;
          }
          setState(() {
            _messages.add({
              'id': DateTime.now().millisecondsSinceEpoch.toString(),
              'from': 'nova',
              'text': reply,
            });
            _updateNovaState(reply, false);
          });
        } else {
          throw Exception("API Error");
        }
      } catch (e) {
        setState(() {
          _messages.add({
            'id': DateTime.now().millisecondsSinceEpoch.toString(),
            'from': 'nova',
            'text': "I'm having a little trouble connecting to my servers. Take a breath — I'm still here with you. (Offline companion mode active)",
          });
          _novaState = NovaState.idle;
        });
      }
    } else {
      await Future.delayed(const Duration(milliseconds: 1200));
      final reply = _getMockResponse(text);
      setState(() {
        _messages.add({
          'id': DateTime.now().millisecondsSinceEpoch.toString(),
          'from': 'nova',
          'text': reply,
        });
        _updateNovaState(reply, false);
      });
    }

    setState(() {
      _loading = false;
    });
    _scrollToEnd();
  }

  void _scrollToEnd() {
    WidgetsBinding.instance.addPostFrameCallback((_) {
      if (_scrollController.hasClients) {
        _scrollController.animateTo(
          _scrollController.position.maxScrollExtent,
          duration: const Duration(milliseconds: 500),
          curve: Curves.easeOutQuart,
        );
      }
    });
  }

  @override
  Widget build(BuildContext context) {
    final isDark = AppColors.isDark(context);

    return Scaffold(
      backgroundColor: Colors.transparent,
      body: Stack(
        children: [
          // Background Gradient Mesh
          Positioned.fill(
            child: AnimatedContainer(
              duration: 2.seconds,
              decoration: BoxDecoration(
                gradient: LinearGradient(
                  begin: Alignment.topCenter,
                  end: Alignment.bottomCenter,
                  colors: [
                    if (_novaState == NovaState.stressed)
                      AppColors.coral.withOpacity(0.05)
                    else if (_novaState == NovaState.happy)
                      AppColors.pink.withOpacity(0.05)
                    else if (_novaState == NovaState.sleeping)
                      AppColors.purpleDark.withOpacity(0.1)
                    else
                      AppColors.purple.withOpacity(isDark ? 0.05 : 0.02),
                    Colors.transparent,
                  ],
                ),
              ),
            ),
          ),

          // Central Nova Avatar
          Positioned(
            top: MediaQuery.of(context).size.height * 0.15,
            left: 0,
            right: 0,
            child: Center(
              child: NovaOrb(state: _novaState, size: 160),
            ),
          ),

          // Foreground Chat
          Column(
            children: [
              // Header
              SafeArea(
                bottom: false,
                child: Padding(
                  padding: const EdgeInsets.symmetric(horizontal: 24, vertical: 16),
                  child: Row(
                    children: [
                      Text(
                        'Nova',
                        style: GoogleFonts.playfairDisplay(
                          fontSize: 28,
                          fontWeight: FontWeight.w600,
                          color: AppColors.text(context),
                        ),
                      ).animate().fadeIn(duration: 400.ms),
                      const Spacer(),
                      Container(
                        padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 6),
                        decoration: BoxDecoration(
                          color: AppColors.purple.withOpacity(0.1),
                          borderRadius: BorderRadius.circular(20),
                        ),
                        child: Row(
                          children: [
                            Container(
                              width: 6,
                              height: 6,
                              decoration: const BoxDecoration(
                                color: AppColors.purpleLight,
                                shape: BoxShape.circle,
                              ),
                            ).animate(onPlay: (controller) => controller.repeat(reverse: true))
                             .fadeIn(duration: 1.seconds),
                            const SizedBox(width: 8),
                            Text(
                              'Online',
                              style: GoogleFonts.inter(
                                fontSize: 12,
                                color: AppColors.purpleLight,
                                fontWeight: FontWeight.w600,
                              ),
                            ),
                          ],
                        ),
                      ),
                    ],
                  ),
                ),
              ),

              // Chat Thread
              Expanded(
                child: ShaderMask(
                  shaderCallback: (Rect bounds) {
                    return const LinearGradient(
                      begin: Alignment.topCenter,
                      end: Alignment.bottomCenter,
                      colors: [Colors.transparent, Colors.white, Colors.white, Colors.white],
                      stops: [0.0, 0.1, 0.9, 1.0],
                    ).createShader(bounds);
                  },
                  child: ListView.builder(
                    controller: _scrollController,
                    physics: const BouncingScrollPhysics(),
                    padding: EdgeInsets.only(
                      top: MediaQuery.of(context).size.height * 0.35, // Push messages down initially
                      left: 16,
                      right: 16,
                      bottom: 24,
                    ),
                    itemCount: _messages.length,
                    itemBuilder: (context, index) {
                      final msg = _messages[index];
                      final isUser = msg['from'] == 'user';
                      return _buildChatBubble(msg['text']!, isUser)
                          .animate().fadeIn(duration: 400.ms).slideY(begin: 0.1, end: 0, curve: Curves.easeOutQuart);
                    },
                  ),
                ),
              ),
              
              // Typing Indicator & Input Area
              ClipRRect(
                child: BackdropFilter(
                  filter: ImageFilter.blur(sigmaX: 20, sigmaY: 20),
                  child: Container(
                    padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 16),
                    decoration: BoxDecoration(
                      color: isDark ? AppColors.bgCard.withOpacity(0.8) : Colors.white.withOpacity(0.8),
                      border: Border(top: BorderSide(color: isDark ? Colors.white.withOpacity(0.05) : AppColors.borderLight)),
                    ),
                    child: SafeArea(
                      top: false,
                      child: Column(
                        children: [
                          if (_loading)
                            Padding(
                              padding: const EdgeInsets.only(bottom: 16.0, left: 16),
                              child: Row(
                                children: [
                                  Text(
                                    "Nova is thinking...",
                                    style: GoogleFonts.inter(
                                      color: AppColors.purpleLight, 
                                      fontSize: 13,
                                      fontWeight: FontWeight.w500,
                                    ),
                                  ).animate(onPlay: (controller) => controller.repeat(reverse: true))
                                   .fade(begin: 0.5, end: 1.0, duration: 1.seconds),
                                ],
                              ),
                            ),
                          Row(
                            children: [
                              Expanded(
                                child: Container(
                                  padding: const EdgeInsets.symmetric(horizontal: 20),
                                  decoration: BoxDecoration(
                                    color: isDark ? Colors.white.withOpacity(0.05) : AppColors.bgElevatedLight,
                                    borderRadius: BorderRadius.circular(30),
                                    border: Border.all(color: isDark ? Colors.white.withOpacity(0.1) : AppColors.borderLight),
                                  ),
                                  child: TextField(
                                    controller: _inputController,
                                    style: GoogleFonts.inter(color: AppColors.text(context)),
                                    onChanged: (text) {
                                      if (text.isNotEmpty && _novaState == NovaState.idle) {
                                        setState(() => _novaState = NovaState.typing);
                                      } else if (text.isEmpty && _novaState == NovaState.typing) {
                                        setState(() => _novaState = NovaState.idle);
                                      }
                                    },
                                    decoration: InputDecoration(
                                      hintText: "Share what's on your mind...",
                                      hintStyle: GoogleFonts.inter(color: AppColors.mutedText(context)),
                                      border: InputBorder.none,
                                      contentPadding: const EdgeInsets.symmetric(vertical: 16),
                                    ),
                                    onSubmitted: (_) => _sendMessage(),
                                  ),
                                ),
                              ),
                              const SizedBox(width: 12),
                              GestureDetector(
                                onTap: () => _sendMessage(),
                                child: Container(
                                  width: 52,
                                  height: 52,
                                  decoration: BoxDecoration(
                                    color: AppColors.purple,
                                    shape: BoxShape.circle,
                                    boxShadow: [
                                      BoxShadow(
                                        color: AppColors.purple.withOpacity(0.3),
                                        blurRadius: 12,
                                        spreadRadius: 2,
                                      ),
                                    ],
                                  ),
                                  child: const Icon(Icons.send_rounded, color: Colors.white, size: 22),
                                ).animate().scale(delay: 200.ms, duration: 300.ms, curve: Curves.easeOutBack),
                              ),
                            ],
                          ),
                        ],
                      ),
                    ),
                  ),
                ),
              ),
            ],
          ),
        ],
      ),
    );
  }

  Widget _buildChatBubble(String text, bool isUser) {
    final isDark = AppColors.isDark(context);
    
    return Padding(
      padding: const EdgeInsets.only(bottom: 24),
      child: Row(
        mainAxisAlignment: isUser ? MainAxisAlignment.end : MainAxisAlignment.start,
        crossAxisAlignment: CrossAxisAlignment.end,
        children: [
          if (!isUser) const SizedBox(width: 8),
          Flexible(
            child: Container(
              padding: const EdgeInsets.symmetric(horizontal: 24, vertical: 18),
              decoration: BoxDecoration(
                color: isUser 
                    ? AppColors.purple 
                    : (isDark ? AppColors.bgGlass : Colors.white),
                borderRadius: BorderRadius.only(
                  topLeft: const Radius.circular(24),
                  topRight: const Radius.circular(24),
                  bottomLeft: Radius.circular(isUser ? 24 : 8),
                  bottomRight: Radius.circular(isUser ? 8 : 24),
                ),
                border: Border.all(
                  color: isUser 
                      ? Colors.transparent 
                      : (isDark ? Colors.white.withOpacity(0.1) : AppColors.borderLight),
                ),
                boxShadow: [
                  BoxShadow(
                    color: Colors.black.withOpacity(0.04),
                    blurRadius: 20,
                    spreadRadius: 0,
                    offset: const Offset(0, 4),
                  )
                ],
              ),
              child: Text(
                text,
                style: GoogleFonts.inter(
                  color: isUser ? Colors.white : AppColors.text(context),
                  fontSize: 16,
                  height: 1.6,
                  fontWeight: FontWeight.w400,
                ),
              ),
            ),
          ),
          if (isUser) const SizedBox(width: 8),
        ],
      ),
    );
  }
}

File: react_native_backup/screens/NovaScreen.js
import React, { useState, useEffect, useRef } from "react";
import {
  StyleSheet,
  View,
  Text,
  TextInput,
  TouchableOpacity,
  ScrollView,
  SafeAreaView,
  KeyboardAvoidingView,
  Platform,
  ActivityIndicator,
} from "react-native";
import Svg, { Circle, Path, G, Defs, RadialGradient, Stop } from "react-native-svg";
import AsyncStorage from "@react-native-async-storage/async-storage";
import { COLORS, FONTS } from "../components/Theme";
import { WarningIcon } from "../components/Icons";

const CRISIS_WORDS = ["suicide", "kill myself", "harm myself", "end my life", "want to die", "self harm"];

function SmilingPlanetAvatar({ size = 90 }) {
  return (
    <Svg width={size} height={size} viewBox="0 0 100 100" fill="none">
      <Defs>
        <RadialGradient id="planetGrad" cx="35%" cy="35%" r="65%">
          <Stop offset="0%" stopColor="#a89af7" />
          <Stop offset="50%" stopColor="#4a3fa0" />
          <Stop offset="100%" stopColor="#1a1040" />
        </RadialGradient>
      </Defs>
      {/* Glow shadow ring */}
      <Circle cx="50" cy="50" r="46" stroke="rgba(168, 154, 247, 0.25)" strokeWidth="1.5" />
      {/* Planet body */}
      <Circle cx="50" cy="50" r="42" fill="url(#planetGrad)" stroke="rgba(168, 154, 247, 0.4)" strokeWidth="1" />
      {/* Outer Orbit Ring */}
      <Path
        d="M 12 58 C 22 75, 78 75, 88 58"
        stroke="rgba(168, 154, 247, 0.7)"
        strokeWidth="2.5"
        strokeLinecap="round"
      />
      {/* Face details */}
      <G opacity="0.9">
        {/* Left Eye */}
        <Circle cx="38" cy="46" r="3.5" fill="white" />
        <Circle cx="37" cy="45" r="1" fill="#1a1040" />
        {/* Right Eye */}
        <Circle cx="62" cy="46" r="3.5" fill="white" />
        <Circle cx="61" cy="45" r="1" fill="#1a1040" />
        {/* Rosy Cheeks */}
        <Circle cx="30" cy="53" r="4" fill="rgba(247, 168, 212, 0.4)" />
        <Circle cx="70" cy="53" r="4" fill="rgba(247, 168, 212, 0.4)" />
        {/* Smile */}
        <Path
          d="M 44 54 Q 50 60, 56 54"
          stroke="white"
          strokeWidth="2"
          strokeLinecap="round"
        />
      </G>
      {/* Front Planet Ring half (to overlay planet) */}
      <Path
        d="M 88 58 C 84 50, 70 42, 50 42 C 30 42, 16 50, 12 58"
        stroke="rgba(168, 154, 247, 0.7)"
        strokeWidth="2.5"
        strokeLinecap="round"
      />
    </Svg>
  );
}

export default function NovaScreen() {
  const [messages, setMessages] = useState([
    {
      id: "1",
      from: "nova",
      text: "I noticed you've been feeling overwhelmed this week. I'm here for you. What would you like to do?",
    },
  ]);
  const [input, setInput] = useState("");
  const [loading, setLoading] = useState(false);
  const [apiKey, setApiKey] = useState("");
  const [safetyTriggered, setSafetyTriggered] = useState(false);

  const scrollViewRef = useRef();

  useEffect(() => {
    async function loadKey() {
      try {
        const storedKey = await AsyncStorage.getItem("zenara_anthropic_api_key");
        if (storedKey) setApiKey(storedKey);
      } catch (e) {
        console.log("Failed to load api key");
      }
    }
    loadKey();
  }, []);

  const checkForCrisis = (text) => {
    const lower = text.toLowerCase();
    return CRISIS_WORDS.some((word) => lower.includes(word));
  };

  const getMockResponse = (userInput) => {
    const lower = userInput.toLowerCase();
    if (lower.includes("reflect")) {
      return "Let's take a moment to look back at your day. What was one moment where you felt completely in control, however small?";
    }
    if (lower.includes("reframe")) {
      return "I can help with that. Share a negative thought you've had today, and we'll work together to identify cognitive distortions and reframe it gently.";
    }
    if (lower.includes("breathe") || lower.includes("breathing")) {
      return "Let's do a quick Box Breathing exercise. Inhale for 4 seconds, hold for 4, exhale for 4, hold for 4. Shall we begin?";
    }
    if (lower.includes("hello") || lower.includes("hi")) {
      return "Hello! I'm Nova, your emotional companion. How is your heart feeling today?";
    }
    return "I hear you, and I'm holding space for you. Thank you for sharing that with me. What feelings are surfacing for you right now?";
  };

  const sendMessage = async (textToSend) => {
    const text = textToSend || input;
    if (!text.trim() || loading || safetyTriggered) return;

    // Safety Trigger Check
    if (checkForCrisis(text)) {
      setSafetyTriggered(true);
      const userMsg = { id: Date.now().toString(), from: "user", text };
      const safetyMsg = {
        id: (Date.now() + 1).toString(),
        from: "nova",
        text: "Safety Warning: I have detected indicators of crisis. The AI companion is suspended. Please see the crisis support information above.",
      };
      setMessages((prev) => [...prev, userMsg, safetyMsg]);
      setInput("");
      return;
    }

    const userMsg = { id: Date.now().toString(), from: "user", text };
    setMessages((prev) => [...prev, userMsg]);
    setInput("");
    setLoading(true);

    if (apiKey) {
      try {
        const res = await fetch("https://api.anthropic.com/v1/messages", {
          method: "POST",
          headers: {
            "Content-Type": "application/json",
            "x-api-key": apiKey,
            "anthropic-version": "2023-06-01",
          },
          body: JSON.stringify({
            model: "claude-3-5-sonnet-20241022",
            max_tokens: 200,
            system: `You are Nova, Zenara's compassionate AI mental health companion. You are warm, empathetic, and supportive. You help users reflect on their emotions, reframe negative thoughts, and practice coping strategies. You are NOT a therapist or medical professional — always remind users to speak with their therapist (Dr. Hayes) for clinical support. Keep responses concise (2-3 sentences), warm, and conversational. Use gentle, supportive language. If you detect any self-harm language, immediately provide crisis resources.`,
            messages: [...messages, userMsg].map((m) => ({
              role: m.from === "user" ? "user" : "assistant",
              content: m.text,
            })),
          }),
        });

        const data = await res.json();
        const reply = data.content?.[0]?.text || "I'm here with you. Tell me more.";
        setMessages((prev) => [...prev, { id: Date.now().toString(), from: "nova", text: reply }]);
      } catch (err) {
        setMessages((prev) => [
          ...prev,
          {
            id: Date.now().toString(),
            from: "nova",
            text: "I'm having a little trouble connecting to my servers. Take a breath — I'm still here with you. (Offline companion mode active)",
          },
        ]);
      }
    } else {
      // Mock Offline Mode
      setTimeout(() => {
        const reply = getMockResponse(text);
        setMessages((prev) => [...prev, { id: Date.now().toString(), from: "nova", text: reply }]);
        setLoading(false);
      }, 800);
      return;
    }
    setLoading(false);
  };

  const handleResetCrisis = () => {
    setSafetyTriggered(false);
    setMessages([
      {
        id: Date.now().toString(),
        from: "nova",
        text: "I'm back, Sarah. Let's start fresh. How can I support you right now?",
      },
    ]);
  };

  return (
    <SafeAreaView style={styles.container}>
      <KeyboardAvoidingView
        behavior={Platform.OS === "ios" ? "padding" : "height"}
        style={styles.keyboardView}
      >
        {/* Header */}
        <View style={styles.header}>
          <Text style={styles.title}>Nova</Text>
          <Text style={styles.subtitle}>AI Companion</Text>
        </View>

        {/* Safety Banner if triggered */}
        {safetyTriggered && (
          <View style={styles.safetyCard}>
            <View style={styles.safetyHeaderRow}>
              <WarningIcon color="#f06c8a" size={18} />
              <Text style={styles.safetyHeader}>IMMEDIATE SUPPORT AVAILABLE</Text>
            </View>
            <Text style={styles.safetyBody}>
              You are not alone. Please reach out to these confidential resources immediately:
            </Text>
            <Text style={styles.safetyContact}>📞 National Helpline (India): 9152987821</Text>
            <Text style={styles.safetyContact}>📞 Suicide & Crisis Lifeline (US): 988</Text>
            <Text style={styles.safetyContact}>📞 Vandrevala Foundation: 1860 2662 345</Text>
            <TouchableOpacity style={styles.safetyResetBtn} onPress={handleResetCrisis}>
              <Text style={styles.safetyResetBtnText}>Reset Companion</Text>
            </TouchableOpacity>
          </View>
        )}

        {/* Nova Glow Moon Avatar */}
        {!safetyTriggered && messages.length <= 2 && (
          <View style={styles.avatarContainer}>
            <View style={styles.glowOuter}>
              <SmilingPlanetAvatar size={90} />
            </View>
          </View>
        )}

        {/* Message Thread */}
        <ScrollView
          ref={scrollViewRef}
          onContentSizeChange={() => scrollViewRef.current?.scrollToEnd({ animated: true })}
          contentContainerStyle={styles.messageScroll}
          showsVerticalScrollIndicator={false}
        >
          {messages.map((msg) => {
            const isUser = msg.from === "user";
            return (
              <View
                key={msg.id}
                style={[
                  styles.messageBubbleContainer,
                  isUser ? styles.bubbleUserAlign : styles.bubbleNovaAlign,
                ]}
              >
                <View
                  style={[
                    styles.messageBubble,
                    isUser ? styles.bubbleUser : styles.bubbleNova,
                  ]}
                >
                  <Text style={styles.messageText}>{msg.text}</Text>
                </View>
              </View>
            );
          })}

          {loading && (
            <View style={[styles.messageBubbleContainer, styles.bubbleNovaAlign]}>
              <View style={[styles.messageBubble, styles.bubbleNova, styles.loadingBubble]}>
                <ActivityIndicator size="small" color={COLORS.purpleLight} />
              </View>
            </View>
          )}
        </ScrollView>

        {/* Suggestion Chips */}
        {!safetyTriggered && messages.length <= 2 && (
          <View style={styles.suggestionsContainer}>
            {[
              "Reflect on my day",
              "Reframe my thoughts",
              "Breathing exercise",
              "Just talk",
            ].map((chip) => (
              <TouchableOpacity
                key={chip}
                style={styles.suggestionChip}
                onPress={() => sendMessage(chip)}
              >
                <Text style={styles.suggestionText}>→ {chip}</Text>
              </TouchableOpacity>
            ))}
          </View>
        )}

        {/* Input Bar */}
        <View style={styles.inputBar}>
          <TextInput
            value={input}
            onChangeText={setInput}
            onSubmitEditing={() => sendMessage()}
            placeholder={safetyTriggered ? "Companion suspended" : "Type your message..."}
            placeholderTextColor={COLORS.muted}
            editable={!safetyTriggered}
            style={[styles.textInput, safetyTriggered && styles.inputDisabled]}
          />
          <TouchableOpacity
            style={[styles.sendButton, safetyTriggered && styles.sendButtonDisabled]}
            disabled={safetyTriggered}
            onPress={() => sendMessage()}
          >
            <Svg width="16" height="16" viewBox="0 0 24 24" fill="none">
              <Path
                d="M5 12h14M12 5l7 7-7 7"
                stroke="white"
                strokeWidth="2.5"
                strokeLinecap="round"
                strokeLinejoin="round"
              />
            </Svg>
          </TouchableOpacity>
        </View>
      </KeyboardAvoidingView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: COLORS.bg,
  },
  keyboardView: {
    flex: 1,
  },
  header: {
    paddingTop: 16,
    paddingBottom: 8,
    alignItems: "center",
    borderBottomWidth: 1,
    borderColor: COLORS.border,
  },
  title: {
    color: COLORS.white,
    fontSize: 20,
    fontWeight: "700",
    fontFamily: FONTS.serif,
  },
  subtitle: {
    color: COLORS.muted,
    fontSize: 12,
    fontFamily: FONTS.sans,
  },
  avatarContainer: {
    alignItems: "center",
    marginTop: 20,
    marginBottom: 10,
  },
  glowOuter: {
    width: 120,
    height: 120,
    borderRadius: 60,
    backgroundColor: "rgba(124, 111, 247, 0.08)",
    alignItems: "center",
    justifyContent: "center",
    shadowColor: COLORS.purple,
    shadowOffset: { width: 0, height: 0 },
    shadowOpacity: 0.6,
    shadowRadius: 20,
    elevation: 4,
  },
  messageScroll: {
    paddingHorizontal: 20,
    paddingVertical: 16,
  },
  messageBubbleContainer: {
    marginBottom: 12,
    flexDirection: "row",
  },
  bubbleUserAlign: {
    justifyContent: "flex-end",
  },
  bubbleNovaAlign: {
    justifyContent: "flex-start",
  },
  messageBubble: {
    maxWidth: "80%",
    borderRadius: 18,
    paddingHorizontal: 16,
    paddingVertical: 12,
    shadowColor: "black",
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 6,
    elevation: 1,
  },
  bubbleUser: {
    backgroundColor: COLORS.purple,
    borderBottomRightRadius: 4,
  },
  bubbleNova: {
    backgroundColor: "rgba(25, 22, 54, 0.75)",
    borderWidth: 1,
    borderColor: COLORS.border,
    borderBottomLeftRadius: 4,
  },
  loadingBubble: {
    paddingVertical: 8,
    paddingHorizontal: 20,
  },
  messageText: {
    color: COLORS.white,
    fontSize: 13.5,
    lineHeight: 19,
    fontFamily: FONTS.sans,
  },
  suggestionsContainer: {
    paddingHorizontal: 20,
    gap: 8,
    marginBottom: 12,
  },
  suggestionChip: {
    backgroundColor: "rgba(25, 22, 54, 0.6)",
    borderRadius: 14,
    borderWidth: 1,
    borderColor: COLORS.border,
    paddingVertical: 10,
    paddingHorizontal: 16,
  },
  suggestionText: {
    color: COLORS.white,
    fontSize: 13,
    fontFamily: FONTS.sans,
  },
  inputBar: {
    flexDirection: "row",
    backgroundColor: "rgba(25, 22, 54, 0.8)",
    borderRadius: 18,
    borderWidth: 1.5,
    borderColor: "rgba(124, 111, 247, 0.25)",
    marginHorizontal: 20,
    marginBottom: 16,
    paddingHorizontal: 14,
    paddingVertical: 8,
    alignItems: "center",
  },
  textInput: {
    flex: 1,
    color: COLORS.white,
    fontSize: 13.5,
    fontFamily: FONTS.sans,
    paddingVertical: 4,
  },
  inputDisabled: {
    color: COLORS.muted,
  },
  sendButton: {
    width: 34,
    height: 34,
    borderRadius: 10,
    backgroundColor: COLORS.purple,
    alignItems: "center",
    justifyContent: "center",
    marginLeft: 8,
    shadowColor: COLORS.purple,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.4,
    shadowRadius: 5,
    elevation: 3,
  },
  sendButtonDisabled: {
    backgroundColor: "rgba(124, 111, 247, 0.2)",
    shadowOpacity: 0,
    elevation: 0,
  },
  safetyCard: {
    backgroundColor: "rgba(240, 108, 138, 0.08)",
    borderColor: "rgba(240, 108, 138, 0.3)",
    borderWidth: 1,
    borderRadius: 16,
    padding: 16,
    margin: 20,
  },
  safetyHeaderRow: {
    flexDirection: "row",
    alignItems: "center",
    gap: 8,
    marginBottom: 8,
  },
  safetyHeader: {
    color: "#f06c8a",
    fontSize: 13,
    fontWeight: "700",
    fontFamily: FONTS.sans,
  },
  safetyBody: {
    color: COLORS.white,
    fontSize: 12,
    lineHeight: 17,
    fontFamily: FONTS.sans,
    marginBottom: 10,
    opacity: 0.9,
  },
  safetyContact: {
    color: COLORS.white,
    fontSize: 12.5,
    fontWeight: "600",
    fontFamily: FONTS.sans,
    marginBottom: 4,
  },
  safetyResetBtn: {
    marginTop: 14,
    backgroundColor: COLORS.bgElevated,
    borderWidth: 1,
    borderColor: COLORS.border,
    borderRadius: 10,
    paddingVertical: 10,
    alignItems: "center",
  },
  safetyResetBtnText: {
    color: COLORS.purpleLight,
    fontSize: 12,
    fontWeight: "600",
  },
});
